fs/expose/dokan/libdokan.py

```diff
@@ -93,154 +102,166 @@ class DOKAN_FILE_INFO(Structure):
-        ("Cleanup", CFUNCTYPE(c_int,
-		LPCWSTR,      # FileName
-		PDOKAN_FILE_INFO)),
-        ("CloseFile", CFUNCTYPE(c_int,
-		LPCWSTR,      # FileName
-		PDOKAN_FILE_INFO)),
-        ("ReadFile", CFUNCTYPE(c_int,
-		LPCWSTR,  # FileName
-		POINTER(c_char),   # Buffer
-		DWORD,    # NumberOfBytesToRead
-		LPDWORD,  # NumberOfBytesRead
-		LONGLONG, # Offset
-		PDOKAN_FILE_INFO)),
-        ("WriteFile", CFUNCTYPE(c_int,
-		LPCWSTR,  # FileName
-		POINTER(c_char), # Buffer
-		DWORD,    # NumberOfBytesToWrite
-		LPDWORD,  # NumberOfBytesWritten
-		LONGLONG, # Offset
-		PDOKAN_FILE_INFO)),
-        ("FlushFileBuffers", CFUNCTYPE(c_int,
-		LPCWSTR,  # FileName
-		PDOKAN_FILE_INFO)),
-        ("GetFileInformation", CFUNCTYPE(c_int,
-		LPCWSTR,  # FileName
-		POINTER(BY_HANDLE_FILE_INFORMATION), # Buffer
-		PDOKAN_FILE_INFO)),
-        ("FindFiles", CFUNCTYPE(c_int,
-		LPCWSTR,  # PathName
-		PFillFindData, # call this function with PWIN32_FIND_DATAW
-		PDOKAN_FILE_INFO)),
-        ("FindFilesWithPattern", CFUNCTYPE(c_int,
-		LPCWSTR,  # PathName
-		LPCWSTR,  # SearchPattern
-		PFillFindData,	#call this function with PWIN32_FIND_DATAW
-		PDOKAN_FILE_INFO)),
-        ("SetFileAttributes", CFUNCTYPE(c_int,
-		LPCWSTR, # FileName
-		DWORD,   # FileAttributes
-		PDOKAN_FILE_INFO)),
-        ("SetFileTime", CFUNCTYPE(c_int,
-		LPCWSTR, # FileName
-		POINTER(FILETIME), # CreationTime
-		POINTER(FILETIME), # LastAccessTime
-		POINTER(FILETIME), # LastWriteTime
-		PDOKAN_FILE_INFO)),
-        ("DeleteFile", CFUNCTYPE(c_int,
-		LPCWSTR, # FileName
-		PDOKAN_FILE_INFO)),
-        ("DeleteDirectory", CFUNCTYPE(c_int,
-		LPCWSTR, # FileName
-		PDOKAN_FILE_INFO)),
-        ("MoveFile", CFUNCTYPE(c_int,
-		LPCWSTR, # ExistingFileName
-		LPCWSTR, # NewFileName
-		BOOL,	 # ReplaceExisiting
-		PDOKAN_FILE_INFO)),
-        ("SetEndOfFile", CFUNCTYPE(c_int,
-		LPCWSTR,  # FileName
-		LONGLONG, # Length
-		PDOKAN_FILE_INFO)),
-        ("SetAllocationSize", CFUNCTYPE(c_int,
-		LPCWSTR,  # FileName
-		LONGLONG, # Length
-		PDOKAN_FILE_INFO)),
-        ("LockFile", CFUNCTYPE(c_int,
-		LPCWSTR,  # FileName
-		LONGLONG, # ByteOffset
-		LONGLONG, # Length
-		PDOKAN_FILE_INFO)),
-        ("UnlockFile", CFUNCTYPE(c_int,
-		LPCWSTR,  # FileName
-		LONGLONG, # ByteOffset
-		LONGLONG, # Length
-		PDOKAN_FILE_INFO)),
-        ("GetDiskFreeSpaceEx", CFUNCTYPE(c_int,
-		PULONGLONG, # FreeBytesAvailable
-		PULONGLONG, # TotalNumberOfBytes
-		PULONGLONG, # TotalNumberOfFreeBytes
-		PDOKAN_FILE_INFO)),
-        ("GetVolumeInformation", CFUNCTYPE(c_int,
-		POINTER(c_wchar),  # VolumeNameBuffer
-		DWORD,	 # VolumeNameSize in num of chars
-		LPDWORD, # VolumeSerialNumber
-		LPDWORD, # MaximumComponentLength in num of chars
-		LPDWORD, # FileSystemFlags
-		POINTER(c_wchar),  # FileSystemNameBuffer
-		DWORD,	 # FileSystemNameSize in num of chars
-		PDOKAN_FILE_INFO)),
-        ("Unmount", CFUNCTYPE(c_int,
-		PDOKAN_FILE_INFO)),
+        ("ZwCreateFile", CFUNCTYPE(NTSTATUS,
```

Comment:
Ok @happy-monk ! you are right only tested on 64-bit !
I made the changes et rebase the pull request 👍